Merge Fixes From Dev-Branch

kapilash opened the pull request:
This PR brings in two fixes in the dev branch that were missing from the master - #6989 and #6991

Also included are pending PR comments from #7031

kapilash commented:
cc @stephentoub @shrutigarg

joshfree commented:
@ericeil can you take a look?

ericeil commented:
Is it possible to submit this as three separate commits: one for each of the two merged commits, and one for the fixes for #7031?  It would make it easier to review now, and easier to track code flow later.

ericeil commented on src/Native/System.Net.Security.Native/pal_gssapi.cpp:
```diff
@@ -35,9 +35,13 @@ static_assert(PAL_GSS_CONTINUE_NEEDED == GSS_S_CONTINUE_NEEDED, "");
 static char gss_spnego_oid_value[] = "\x2b\x06\x01\x05\x05\x02"; // Binary representation of SPNEGO Oid (RFC 4178)
-static gss_OID_desc gss_mech_spnego_OID_desc = {.length = 6, .elements = static_cast<void*>(gss_spnego_oid_value)};
+static gss_OID_desc gss_mech_spnego_OID_desc = {
+    .length = static_cast<OM_uint32>(sizeof(gss_spnego_oid_value) / sizeof(gss_spnego_oid_value[0]) - 1),
```
It looks like we have an `ARRAY_SIZE` macro that would make this more clear:

```
    .length = ARRAY_SIZE(gss_spnego_oid_value) - 1,
```

kapilash replied:
done. Thank you. Did not know of the macro

ericeil commented on src/Native/System.Net.Security.Native/pal_gssapi.cpp:
```diff
@@ -35,9 +35,13 @@ static_assert(PAL_GSS_CONTINUE_NEEDED == GSS_S_CONTINUE_NEEDED, "");
 static char gss_ntlm_oid_value[] = "\x2b\x06\x01\x04\x01\x82\x37\x02\x02\x0a"; // Binary representation of NTLM OID (https://msdn.microsoft.com/en-us/library/cc236636.aspx)
-static gss_OID_desc gss_mech_ntlm_OID_desc = { .length = 10, .elements = static_cast<void*>(gss_ntlm_oid_value) };
+static gss_OID_desc gss_mech_ntlm_OID_desc = { 
+    .length = static_cast<OM_uint32>(sizeof(gss_ntlm_oid_value) / sizeof(gss_ntlm_oid_value[0]) - 1),
```
Use `ARRAY_SIZE` here as well.

ericeil commented:
Aside from a couple of nits, this LGTM.  @stephentoub or @bartonjs might have more context, and should also have a look.

kapilash commented:
> Is it possible to submit this as three separate commits: one for each of the two merged commits, and one for the fixes for #7031? It would make it easier to review now, and easier to track code flow later.

Sure. Rebased it with master and converted it to three commits.

stephentoub commented:
Test Outerloop RHEL7.2 Debug please
Test Outerloop Ubuntu14.04 Debug please

stephentoub commented:
LGTM

kapilash commented:
Fixed the typo in the commit message.. Innerloop OSX failures are due to 

>  Build machine went offline during the build.

Will launch again

kapilash commented:
@dotnet-bot 
test innerloop OSX DEBUG please
@dotnet-bot test innerloop OSX RELEASE please

kapilash commented:
Test Innerloop OSX Debug please

kapilash commented:
Test Outerloop RHEL7.2 Debug please
Test Outerloop Ubuntu14.04 Debug please

stephentoub commented:
> Fixed the typo in the commit message

I think you meant "Pending PR comments" not "Pending PR commits".  But it's fine.  Not worth restarting CI for.

kapilash commented:
> I think you meant "Pending PR comments" not "Pending PR commits".

oops. There was another typo -  it was "prending" which I fixed. 
Thanks for review
